lib/Runtime/Base/ExpirableObject.cpp

```diff
@@ -28,17 +27,38 @@ void ExpirableObject::Dispose(bool isShutdown)
 {
 }
 
+void * ExpirableObject::GetRegistrationHandle()
+{
+    return (void *)(registrationHandle & ~1);
+}
+
+void 
+ExpirableObject::SetRegistrationHandle(void * registrationHandle)
+{
+    Assert(this->GetRegistrationHandle() == nullptr);
+    Assert(((intptr_t)registrationHandle & 1) == 0);
+    Assert(registrationHandle != nullptr);
+    this->registrationHandle |= (intptr_t)registrationHandle;
+}
+
+void
+ExpirableObject::ClearRegistrationHandle()
+{
+    Assert(this->GetRegistrationHandle() != nullptr);
```

Comment:
If the expirable object is used, this assert wont catch the double free case, since the handle would be 1? Why not just set the handle to nullptr in this function always? Because we want to consider it used if Clear was called?

Reply:
Theortically we can set it to null probably , I just want the function to do one thing (like what it did before), and I don't want to change the behavior by the fact that we encoded the isUsed field at the same location.